### PR TITLE
Update to Django 1.7

### PR DESCRIPTION
**Pull request by frankh**
Main changes are to migrations.

I started them again from scratch and then renamed check to status_check (as check is now reserved by django).

The plugins also needed updating to be django 1.7 compatible. (they have to defined their own migrations if they depend on an app with migrations now)

### CODE REVIEW

**Comment by frankh** [comment]
Annoyingly PR is quite noisy because it removes all the migrations, other than that it's pretty small

**Comment by michalc** [comment]
Should the check -> status_check be done before the upgrade, as a separate PR, with its own (old-style) migration? (Just to limit scope of the actual upgrade)

Also: restarting migrations. Would this mean that people wouldn't be able to upgrade multiple versions of cabot in one go? In some cases, they would have to upgrade to the last one of the old migration system, and then again to the version they want? Is there some way to convert the migrations?

**Comment by michalc** [comment]
In https://github.com/arachnys/cabot/pull/411/commits/73cdef5f45827c3f88c9a699ac7821fe6fef71b1 there are a couple of changes that don't seem to be explained by the commit message. Separate out to separate commits? (And even PRs if they can be merged in ahead of the upgrade...)

**Comment by frankh** [comment]
good call, moved them into https://github.com/arachnys/cabot/pull/413

File: cabot/cabotapp/tests/tests_basic.py [hunk 1]
```diff
@@ -932,6 +934,16 @@ def test_alert(self, fake_send_alert):
         self.assertEqual(fake_send_alert.call_count, 1)
         fake_send_alert.assert_called_with(self.service, duty_officers=[])
 
+    def test_update_plugins(self):
+        # Test that disabling a plugin is detected by update_alert_plugins
+        plugins = update_alert_plugins()
+        plugin_count = len(plugins)
+        new_apps = [s for s in settings.INSTALLED_APPS if s not in ['cabot_alert_hipchat']]
```
Comment (michalc) [inline cabot/cabotapp/tests/tests_basic.py]: Why not hipchat?

File: cabot/cabotapp/views.py [hunk 2]
```diff
@@ -704,7 +704,9 @@ class ServiceCreateView(LoginRequiredMixin, CreateView):
     model = Service
     form_class = ServiceForm
 
-    alert.update_alert_plugins()
+    def __init__(self, *args, **kwargs):
+        alert.update_alert_plugins()
+        super(ServiceCreateView, self).__init__(*args, **kwargs)
```
Comment (michalc) [inline cabot/cabotapp/views.py]: Is this the code that the test in the separate commit tests? Can this and that test be in one commit, and in one PR if it's current django version compatible?

Comment (frankh) [inline cabot/cabotapp/views.py]: yep, just did that https://github.com/arachnys/cabot/pull/413

File: cabot/cabotapp/tests/tests_basic.py (hunk 1 above)
Comment (frankh) [inline cabot/cabotapp/tests/tests_basic.py]: just picked a random one. It's just to make sure it updates properly after a plugin is removed

**Comment by frankh** [comment]
> Also: restarting migrations. Would this mean that people wouldn't be able to upgrade multiple versions of cabot in one go? In some cases, they would have to upgrade to the last one of the old migration system, and then again to the version they want? Is there some way to convert the migrations?

AFAIK there is no way to convert them

I think django may be smart enough to partially apply migrations now though

File: cabot/cabotapp/views.py (hunk 2 above)
Comment (frankh) [inline cabot/cabotapp/views.py]: nevermind, test didn't work properly on 1.6

**Comment by michalc** [comment]
> I think django may be smart enough to partially apply migrations now though

Can you clarify?

Also: regarding upgrade paths, if there is no way to transfer the migrations over, then should there be some documentation as to how to upgrade to this version, say in the README or at http://cabotapp.com/use/deployment.html ?

**Comment by frankh** [comment]
> Can you clarify?

Not really, looked into it and can't find anything on it. Might have just made it up

Docs are definitely a good idea but they aren't part of this repo so need to be updated separately.

File: cabot/cabotapp/tests/tests_basic.py (hunk 1 above)
Comment (michalc) [inline cabot/cabotapp/tests/tests_basic.py]: Ah, I think I get it now

**Comment by michalc** [comment]
Also: check -> status_check... this means that it's essentially a breaking change of the server API? Are we, as in the code in this repository, the only consumer of this? I wonder if this needs a bit of investigation... There could be open/closed source projects using it?

Is there no workaround to keep it as "check"?

**Comment by frankh** [comment]
It only changes the schema, no changes to any api's.

Also I can't imagine anyone would have built anything relying on models... Possibly some cabot plugins, but they will need fixing for django 1.7 anyway most likely

**Comment by dbuxton** [comment]
@michalc it's a good point but in fact the StatusCheckResult model isn't exposed by any api (see https://github.com/arachnys/cabot/blob/master/cabot/rest_urls.py) and so changes there won't break external clients.

**Comment by JeanFred** [comment]
> Docs are definitely a good idea but they aren't part of this repo so need to be updated separately.

They are in the `gh-pages` branch − but that does need to be updated separately indeed.

**Pull request merged**